owncloudComLibrary/src/main/java/com/owncloud/android/lib/common/OwnCloudClient.java

```diff
@@ -102,8 +97,17 @@ public int executeHttpMethod(HttpBaseMethod method) throws Exception {
         int status;
 
         do {
-            setRequestId(method);
-
+            String requestId = RandomUtils.generateRandomUUID();
+
+            // Header to allow tracing requests in apache and ownCloud logs
+            Timber.d("Executing in request with id %s", requestId);
+            method.setRequestHeader(HttpConstants.OC_X_REQUEST_ID,requestId);
+            method.setRequestHeader(HttpConstants.USER_AGENT_HEADER, SingleSessionManager.getUserAgent());
+            method.setRequestHeader(HttpConstants.PARAM_SINGLE_COOKIE_HEADER, "true");
+            method.setRequestHeader(HttpConstants.ACCEPT_ENCODING_HEADER, HttpConstants.ACCEPT_ENCODING_IDENTITY);
```

Comment:
What about creating a new `setRequestHeaders` method with variable args and write something like

```
method.setRequestHeaders(
    Pair(HttpConstants.OC_X_REQUEST_ID, requestId)
    Pair(HttpConstants. USER_AGENT_HEADER, SingleSessionManager.getUserAgent())
)
```

or even with a new RequestHeader object

```
method.setRequestHeaders(
    RequestHeader(HttpConstants.OC_X_REQUEST_ID, requestId)
    RequestHeader(HttpConstants. USER_AGENT_HEADER, SingleSessionManager.getUserAgent())
)
```

It's just a suggestion 🍺

Reply:
Could be an option, but i didn't want to perform a big refactor. We could include that when we convert OwnCloudClient to kotlin.